geomesa-features/geomesa-feature-avro-complex/src/main/scala/org/locationtech/geomesa/features/avro/complex/AvroCodecs.scala

```diff
@@ -0,0 +1,704 @@
+package org.locationtech.geomesa.features.avro.complex
+
+import java.nio.ByteBuffer
+import java.sql.Timestamp
+import java.{net, sql, util}
+import java.util.function.Function
+import java.util.{Calendar, Date, GregorianCalendar, UUID, ArrayList => JArrayList, Collection => JCollection, Collections => JCollections, HashMap => JHashMap}
+
+import com.vividsolutions.jts.geom.Geometry
+import com.vividsolutions.jts.io.{WKBReader, WKBWriter}
+import org.apache.avro.Schema
+import org.apache.avro.io.{Decoder, Encoder}
+import org.geotools.feature.`type`.AttributeDescriptorImpl
+import org.geotools.feature.{AttributeImpl, ComplexAttributeImpl, FeatureImpl}
+import org.geotools.filter.identity.FeatureIdImpl
+import org.geotools.gml3.GMLSchema
+import org.locationtech.geomesa.features.avro.complex.AvroSchemas.{FieldNameEncoder, _}
+import org.locationtech.geomesa.utils.geotools.RichAttributeDescriptors._
+import org.opengis.feature.`type`.{PropertyDescriptor, _}
+import org.opengis.feature.{Attribute, ComplexAttribute, Property}
+import org.opengis.filter.identity.FeatureId
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+object AvroCodecs {
+
+  val primitiveTypes =

```

Comment:
This bit of code closely mimics https://github.com/locationtech/geomesa/blob/master/geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureUtils.scala#L90 and I think ideally within geomesa a lot of this code should be shared with the simple feature serialization?